src/main/java/com/sap/oss/phosphor/fosstars/data/github/UsesSnyk.java

```diff
@@ -0,0 +1,212 @@
+      if (content.isPresent() && content.get().length() >= ACCEPTABLE_CONFIG_SIZE) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if a repository has a policy file for Snyk.
+   *
+   * @param repository The repository
+   * @return True if a policy file was found, false otherwise.
+   */
+  private boolean hasSnykPolicy(LocalRepository repository) throws IOException {
+    List<Path> snykPolicyFilePaths = repository.files(SNYK_FILE_PREDICATE);
+    return !snykPolicyFilePaths.isEmpty();
+  }
+
+  /**
+   * Checks whether a project has open pull requests from Snyk.
+   *
+   * @param project The project.
+   * @return True if the project has open pull requests form Snyk.
+   * @throws IOException If something went wrong.
+   */
+  private boolean hasOpenPullRequestFromSnyk(GitHubProject project) throws IOException {
+    return fetcher.repositoryFor(project).getPullRequests(GHIssueState.OPEN).stream()
+        .anyMatch(this::createdBySnyk);
+  }
+
+  /**
+   * Checks if a pull request was created by Snyk.
+   *
+   * @param pullRequest The pull request.
+   * @return True if the user looks like Snyk, false otherwise.
+   */
+  private boolean createdBySnyk(GHPullRequest pullRequest) {
+    try {
+      GHUser user = pullRequest.getUser();
+      return isSnyk(user.getName()) || isSnyk(user.getLogin());
+    } catch (IOException e) {
+      logger.warn("Oops! Could not fetch name or login!", e);
+      return false;
+    }
+  }
+
+  /**
+   * Checks if a repository contains commits from Snyk in the commit history.
+   *
+   * @param repository The repository.
+   * @return True if at least one commit from Snyk was found, false otherwise.
+   */
+  private boolean hasSnykCommits(LocalRepository repository) {
+    Date date = Date.from(Instant.now().minus(ONE_YEAR));
+
+    try {
+      for (Commit commit : repository.commitsAfter(date)) {
+        if (isSnyk(commit)) {
+          return true;
+        }
+      }
+    } catch (IOException e) {
+      logger.warn("Something went wrong!", e);
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if a commit was done by Snyk.
+   *
+   * @param commit The commit to be checked.
+   * @return True if the commit was done by Snyk, false otherwise.
+   */
+  private static boolean isSnyk(Commit commit) {
+    if (isSnyk(commit.authorName()) || isSnyk(commit.committerName())) {
+      return true;
+    }
+
+    for (String line : commit.message()) {
+      if ((line.startsWith("Signed-off-by:") || line.startsWith("Co-authored-by:"))
+          && line.contains(SNYK_PATTERN)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks whether a name looks like Snyk.
+   *
+   * @param name The name.
+   * @return True if the name looks like Snyk, false otherwise.
+   */
+  private static boolean isSnyk(String name) {
+    return name != null && name.toLowerCase().contains(SNYK_PATTERN);
+  }
+}
```

Comment:
I believe most of the methods given here seems to be from `UsesDependebot`. Could you please provide an `abstract class`?

Reply:
Done